src/native/minipal/log.c

```diff
@@ -280,25 +293,47 @@ static int sync_file(minipal_log_flags flags)
 }
 #endif
 
-static ssize_t write_file(int fd, const char* msg, size_t bytes_to_write)
+static ssize_t write_file(minipal_log_flags flags, const char* msg, size_t bytes_to_write)
 {
     ssize_t ret = 0;
-    while ((ret = write(fd, msg, bytes_to_write)) < 0 && errno == EINTR);
+    while ((ret = write(fileno(get_std_file(flags)), msg, bytes_to_write)) < 0 && errno == EINTR);
     return ret;
 }
 #endif
 
+typedef ssize_t (*write_file_fnptr)(minipal_log_flags flags, const char* msg, size_t bytes_to_write);
+
 int minipal_log_write(minipal_log_flags flags, const char* msg)
 {
     assert(msg != NULL && msg[0] != '\0');
 
-    size_t bytes_to_write = strlen(msg);
+    size_t bytes_to_write = 0;
     size_t bytes_written = 0;
 
-    int fd = fileno(get_std_file(flags));
+    write_file_fnptr write_fnptr = write_file;
+
+#ifdef HOST_WINDOWS
+    const char* msg_char = msg;
+    while (*msg_char)
```

Comment:
A nit - you can use `msg_char = strstr(msg, "\r\n")` for a shorter code

Reply:
Its actually intentional to take strlen as well as check for the pattern in one loop over the string. If we used strstrs we would need to do strlen as well, meaning two passes over the string, so this does both in one pass.